tests/checkpointing/test_legacy_checkpoints.py

```diff
@@ -14,84 +14,118 @@
 import glob
 import os
 import sys
+from unittest.mock import patch
 
 import pytest
+import torch
 
-from pytorch_lightning import Trainer
+import pytorch_lightning as pl
+from pytorch_lightning import Callback, Trainer
+from pytorch_lightning.callbacks import EarlyStopping
 from tests import _PATH_LEGACY
 
 LEGACY_CHECKPOINTS_PATH = os.path.join(_PATH_LEGACY, 'checkpoints')
 CHECKPOINT_EXTENSION = ".ckpt"
+LEGACY_BACK_COMPATIBLE_PL_VERSIONS = (
```

Comment:
I have not populated all past legacy, so all legacy test are failing as they are expecting the actual model
once I do it and upload to S3, merging this PR become p0 - blocker